Converters/ArcGIS/Speckle.Converters.ArcGIS3/ToSpeckle/Raw/MultipatchFeatureToSpeckleConverter.cs

```diff
@@ -44,83 +42,127 @@ public IReadOnlyList<Base> Convert(ACG.Multipatch target)
+    List<int> faces = new();
+    List<double> vertices = new();
+
+    for (int i = 0; i < points.Count; i++)
+    {
+      if (i >= 2) // every new point adds a triangle
       {
-        // first ring means a start of a new PolygonGeometry3d
-        List<double> pointCoords = allPoints[idx].SelectMany(x => new List<double>() { x.x, x.y, x.z }).ToList();
-        List<int> faces = new() { pointCoords.Count };
-        faces.AddRange(Enumerable.Range(0, pointCoords.Count));
-        SOG.Mesh multipatch =
-          new()
-          {
-            vertices = pointCoords,
-            faces = faces,
-            units = _settingsStore.Current.SpeckleUnits
-          };
-        converted.Add(multipatch);
+        faces.AddRange(new List<int>() { 3, i - 2, i - 1, i });
+        vertices.AddRange(pointCoords.GetRange(3 * (i - 2), 9).ToList());
       }
-      /* ignore before we support triangulation, then only add shape after looping through all loops: if (idx == target.PartCount - 1)
-      else if (patchType == ACG.PatchType.Ring)
+    }
+
+    return new()
+    {
+      faces = faces,
+      vertices = vertices,
+      units = _settingsStore.Current.SpeckleUnits
+    };
+  }
+
+  private SOG.Mesh GetMeshFromTrianglesPatch(List<ACG.MapPoint> points)
+  {
+    List<double> pointCoords = points.SelectMany(x => new List<double>() { x.X, x.Y, x.Z }).ToList();
+    List<int> faces = new();
+    List<double> vertices = new();
+
+    for (int i = 0; i < points.Count; i++)
+    {
+      if (i >= 2 && (i + 1) % 3 == 0) // every 3 new points is a new triangle
       {
-        List<double> pointCoords = allPoints[idx].SelectMany(x => new List<double>() { x.x, x.y, x.z }).ToList();
-        SOG.Polyline polyline = new() { value = pointCoords, units = _settingsStore.Current.SpeckleUnits };
-
-        // every outer ring is oriented clockwise
-        bool isClockwise = polyline.IsClockwisePolygon();
-        if (!isClockwise)
-        {
-          // add void to existing polygon
-          polygonGeom.voids.Add(polyline);
-        }
-        else
-        {
-          // add existing polygon to list, start a new polygon with a boundary
-          converted.Add(polygonGeom);
-          polygonGeom = new()
-          {
-            voids = new List<ICurve>(),
-            boundary = polyline,
-            units = _settingsStore.Current.SpeckleUnits
-          };
-        }
-        // if it's already the last part, add to list
-        if (idx == target.PartCount - 1)
-        {
-          converted.Add(polygonGeom);
-        }
+        faces.AddRange(new List<int>() { 3, i - 2, i - 1, i });
+        vertices.AddRange(pointCoords.GetRange(3 * (i - 2), 9).ToList());
       }
-      */
-      else
+    }
+
+    return new()
+    {
+      faces = faces,
+      vertices = vertices,
+      units = _settingsStore.Current.SpeckleUnits
+    };
+  }
+
+  private SOG.Mesh GetMeshFromTriangleFanPatch(List<ACG.MapPoint> points)
+  {
```

Comment:
I want to know what are they and `from what to what` we are trying to convert. mostly it is a question for @KatKatKateryna. could you document these scenarios on figma by giving some examples with shapes or real scenarios?